### ISSUE TITLE
Hedrot receiver / freeze max

### ISSUE THREAD

**Issue by DanaGuen** [abaskind/hedrot#1]
Bonjour,
Dans le cadre de mon mémoire de master je travaille sur le binaural et ai voulu utiliser hedrot avec le plugin Mybino. J'ai donc assemblé les composants, suivi scrupuleusement le manuel d'utilisation afin d'installer les drivers nécessaires...mais dès que j'ouvre le receiver (que ce soit le .exe ou le patch), max freeze. Je suis sur pc win10 64 bits et utilise max 7 (64 bits également). J'ai tenté d'installer une version précédente, même problème. Que faire? (autre que d'aller chercher un mac si possible)
Merci d'avance,
Dana

Hi,
For my dissertation I'm working on binaural audio and I wanted to use hedrot with the Mybino plugin. I've assembled the headtracker and followed all the instructions concerning the software requirements...but each time I open the receiver (.exe or max patch), max freezes. I'm using a windows 10 64 bits pc and max 7 (x64). I tried to run a previous version of hedrot, same problem. What can I do? (which doesn't include to use a mac if possible)
Thanks,
Dana

**Comment by abaskind** [comment]
Hi Dana,

it looks like an issue with shared libs. Could you confirm me that you did the following steps?
1/ install Microsoft Visual C++ 2012 Redistributable package version x64
2/ add the folder "dll" to the windows path folder (either user or system)

Best,

Alexis

**Comment by DanaGuen** [comment]
Hi and thanks for answering!
I can join you screenshots, I already had Microsoft Visual.
So I guess it should be the path folder I chose (I'll try another one).
I hope I'll get it work! :)

Dana

2018-01-15 18:23 GMT+01:00 Alexis Baskind <notifications@github.com>:

**Comment by DanaGuen** [comment]
(...looks like the problem remains, I first added the path in the user
section; no bug, but the hedrotreceiver, central object of the patch, can't
be recognized by max, so it's useless. If I put manually all dll files in
the WindowsApps folder (as shown just before), the object is accessible in
max, but the latter says "no headtracker connected" and then crashes. So
I've erased the dll files of the WindowsApps folder and the newly created
path in the user variable environment. And tried this time with the system
path folder. Same problem...)


2018-01-15 22:55 GMT+01:00 Danae Le Guennic <d.leguennic@gmail.com>:

**Comment by abaskind** [comment]
Hi Danae,

> (...looks like the problem remains, I first added the path in the user section; no bug, but the hedrotreceiver, central object of the patch, can't be recognized by max, so it's useless. If I put manually all dll files in the WindowsApps folder (as shown just before), the object is accessible in max, 

... which may mean that the path was not correct previously, otherwise the behavior should not change if you leave the dlls in the folder and add the folder to the path, or if you put the dlls in WindowsApps.

Could you test if the path was entered correctly? For instance put a .exe in the same folder, open a console (of course not in this folder) and type the name of this program to see if it recognizes it. Do your path name has weird characters or spaces?

On top of that: I'm not 100% sure that WindowsApps is the right place for dll from 3rd party dlls, since it's meant for windows store programs.

Two last clues for today: 
1/ have you installed the Windows Serial installer from Teensy?
https://www.pjrc.com/teensy/td_download.html
2/ test the command-line version of hedrot (hedrotReceiverDemo.exe) in a console.

Best,

Alexis

**Comment by DanaGuen** [comment]
Hi,
I join you more screenshots.
At each try I deleted the former one, to be sure of what was influencing
the result. I loaded the firmware on my tennsy and kept the teensy app
opened, then ran the hedrotreceiver.exe.
(1.jpg) I tried to add the dll folder path (I verified, it's correctly
written) in the "Path" variable in *user environment* where there already
was WindowsApps. Max couldn't recognize the library (let's call it Problem
1), so I tried to manually copy the dlls inside the WindowsApps folder
(just to see if it would be any different). Indeed this time the max object
was available, but max crashed (Problem 2). NB: when I launched the
headtracking function, it said "candidate found, waiting for confirm", but
was eventually unable to stick to it (no headtracker found).
(2.jpg) I tried to add the dll folder path in the "Path" variable in *system
environment*: Problem 2.
(2b.jpg) Tried to create a new variable in *system environment: *Problem 2.
Tried then to run the demo (demo.jpg). Does it help?

I do have the teensy software, I think the problem remains in the dlls
:/...could my java be a problem?
Thanks a lot for helping! :)


2018-01-16 22:09 GMT+01:00 Alexis Baskind <notifications@github.com>:
